packages/main/src/plugin/onboarding-registry.spec.ts

```diff
@@ -47,131 +48,213 @@ const context = new Context(apiSender);
+    expect(onboarding?.steps[0].status).toBe('completed');
+    expect(context.getValue(contextKey)).toBe('test');
+    // reset all states
+    onboardingRegistry.resetOnboarding([extensionId]);
+    // check states have been reset
+    onboarding = onboardingRegistry.getOnboarding(extensionId);
+    expect(onboarding).toBeDefined();
+    expect(onboarding?.status).toBe(undefined);
+    expect(onboarding?.steps[0].status).toBe(undefined);
+    expect('test' in context.collectAllValues()).toBe(false);
+  });
+
+  test('Should throw if no onboarding for that extension', async () => {
+    expect(() => onboardingRegistry.resetOnboarding(['unknown'])).toThrowError(
+      'No onboarding found for extensions unknown',
+    );
+  });
 });
 
-test('Should throw if no onboarding for that extension', async () => {
-  expect(() => onboardingRegistry.resetOnboarding(['unknown'])).toThrowError(
-    'No onboarding found for extensions unknown',
+test('checkIdsReadability should detect non valid ids', () => {
+  const onboardingRegistry = new OnboardingRegistry(configurationRegistry, context);
+  const extensionPath = '/root/path';
+  const extension = {
+    path: extensionPath,
+    id: extensionId,
+  } as AnalyzedExtension;
+  const onboarding = {
+    title: 'Get started with Podman Desktop',
+    steps: [
+      {
+        id: 'welcomeViewNotOK',
+        label: 'Check Podman',
+        title: 'Checking for Podman installation',
+      },
+      {
+        id: 'welcomeView',
+        label: 'Check Podman',
+        title: 'Checking for Podman installation',
+      },
+      {
+        id: 'checkInstalledCommandNotOK',
+        label: 'Check Podman',
+        title: 'Checking for Podman installation',
+        command: 'podman.onboarding.checkPodmanInstalled',
+        completionEvents: ['onCommand:podman.onboarding.checkPodmanInstalled'],
+      },
+      {
+        id: 'checkInstalledCommand',
+        label: 'Check Podman',
+        title: 'Checking for Podman installation',
+        command: 'podman.onboarding.checkPodmanInstalled',
+        completionEvents: ['onCommand:podman.onboarding.checkPodmanInstalled'],
+      },
+      {
+        id: 'installFailure',
+        label: 'Installation failed',
+        title: 'Installation failed',
+        state: 'failed' as OnboardingState,
+      },
+      {
+        id: 'installFailureNotOK',
+        label: 'Installation failed',
+        title: 'Installation failed',
+        state: 'failed' as OnboardingState,
+      },
+      {
+        id: 'installSuccess',
+        label: 'Installation successful',
+        title: 'Installation successful',
+        state: 'completed' as OnboardingState,
+      },
+      {
+        id: 'installSuccessNotOK',
+        label: 'Installation successful',
+        title: 'Installation successful',
+        state: 'completed' as OnboardingState,
+      },
+    ],
+    enablement: 'true',
+  };
+
+  const consoleWarnMock = vi.fn();
+  console.warn = consoleWarnMock;
```

Comment:
IMHO it should be part of a beforeEach, afterEach where in after step we reset to the previous console.warn object to not mess up further tests